src/curl.c

```diff
@@ -69,9 +75,115 @@ int swupd_curl_init(void)
 		return -1;
 	}
 
+#ifdef FALLBACK_CAPATHS
+	/* parse and initialize CA paths. */
+	{
+		char *str = strdup(FALLBACK_CAPATHS);
+		char *tok;
+		char *ctx;
+		struct stat st;
+		int i = 0;
+
+		tok = strtok_r(str, ":", &ctx);
+		do {
+			if (stat(tok, &st)) {
+				continue;
+			}
+			if ((st.st_mode & S_IFMT) != S_IFDIR) {
+				continue;
+			}
+			fallback_capaths[i] = strdup(tok);
+			fprintf(stderr, "Alternative CApath: %s\n", fallback_capaths[i]);
+			if (++i > CA_PATH_NO_MAX) {
+				fprintf(stderr, "Warning: up to %d CApaths can be specified."
+						" Ignoring the rest.\n", CA_PATH_NO_MAX);
+				break;
+			}
+
+		} while ((tok = strtok_r(NULL, ":", &ctx)));
+		fallback_capaths_no = i;
+		free(str);
+	}
+#endif
+
 	return 0;
 }
 
+static void swupd_curl_test_resume(void);
+
+int swupd_curl_check_network(void) {
+	CURLcode curl_ret;
+	int ret = -1;
+	int i = -1;
+	CURL *c;
+	static int has_network = 0;
+
+	if (has_network) return 0;
+
+	if (!curl) {
+		return -1;
+	}
+
+	c = curl_easy_duphandle(curl);
+	if (!c) {
+		return -1;
+	}
+
+	/* cycle starts with -1 which tests the default */
+	while (i < fallback_capaths_no) {
+		curl_easy_reset(c);
+
+		curl_ret = curl_easy_setopt(c, CURLOPT_URL, VERSIONURL);
```

Comment:
Compilation fails here because VERSIONURL is undeclared

Reply:
Hm. Why travis configures a broken swupd? :) Without any endpoints configured. Sorry, will fix this...